accord-core/src/main/java/accord/utils/random/Picker.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.utils.random;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.IntSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+import accord.utils.Invariants;
+import accord.utils.RandomSource;
+
+public class Picker
```

Comment:
This was all to clean up code I wasn't fond of in the prior patch that has since been removed, so I will remove it. However, we should also entirely remove `RandomSource.pickX()` because it isn't used in this codebase, only within C* (in places where we have alternatives).

If we need it in future, we should wrangle over how to best design it at that date - if you need to introduce it, ping me and we'll discuss it then?

Reply:
I don't have issues with `pick*` methods moving to `org.apache.cassandra.utils.AccordGenerators` and living in C* for now.  The exampled used in accord are in pending work not in review stage yet, so that is fine by me.

Reply:
I've kept just the WeightedRandomPicker and updated its API a little, since I use these